manim/scene/scene_file_writer.py

```diff
@@ -40,6 +40,55 @@
     from manim.renderer.opengl_renderer import OpenGLRenderer
 
 
+def to_av_frame_rate(fps):
+    epsilon1 = 1e-4
+    epsilon2 = 0.02
+
+    if isinstance(fps, int):
+        (num, denom) = (fps, 1)
+    elif abs(fps - round(fps)) < epsilon1:
+        (num, denom) = (round(fps), 1)
+    else:
+        denom = 1001
+        num = round(fps * denom / 1000) * 1000
+        if abs(fps - num / denom) >= epsilon2:
+            raise ValueError("invalid frame rate")
+
+    return av.utils.Fraction(num, denom)
+
+
+def convert_audio(input_path: Path, output_path: Path, codec_name: str):
+    with (
+        av.open(input_path) as input_audio,
+        av.open(output_path, "w") as output_audio,
+    ):
+        input_audio_stream = input_audio.streams.audio[0]
+        output_audio_stream = output_audio.add_stream(codec_name)
+        for frame in input_audio.decode(input_audio_stream):
+            for packet in output_audio_stream.encode(frame):
+                output_audio.mux(packet)
+
+        for packet in output_audio_stream.encode():
+            output_audio.mux(packet)
+
+
+def get_out_format(path: Path | str, check_config=False) -> str:
+    if isinstance(path, str):
+        path = Path(path)
+
+    suffix = path.suffix.lower()
+    if not suffix.startswith("."):
+        raise ValueError(f"{path}: no suffix")
+
+    out_format = suffix[1:]
+    if check_config and config.format is not None and config.format != out_format:
```

Comment:
Maybe I'm misunderstanding, but wouldn't this fail for `config.transparent = True` and `config.format = anything_but_webm`?

I think the best way to get the out format is to just use `config.move_file_extension.removeprefix(".")` - and then you can also avoid the checks.

Reply:
I agree with this; I've pushed a version of this where the config is treated as the single source of truth in 1835c2a5. (I checked that everywhere `combine_files` and friends is called, the file suffix eventually comes out of the config anyways.)